File: program_data/github_java_sort_function/10/90.java
public int[] bucketSort(int[] num, int bucketSize){
        if(num.length == 0 || num.length == 1) return num;
        
        int max = Integer.MIN_VALUE;
        int min = Integer.MAX_VALUE;
        for(int i = 0; i < num.length; i++){
            if(max < num[i]){
                max = num[i];
            }
            if(min > num[i]){
                min = num[i];
            }
        }
        System.out.println("max:" + max + ";" + " min:" + min);
        int bucketCount = ((max - min)%bucketSize == 0)? (max - min)/bucketSize :(max - min)/bucketSize +1  ;
        
        
        List<List<Integer>> intermediate = new ArrayList<List<Integer>>(bucketCount);
        System.out.println("bucketCount: " + bucketCount);

        for(int i = 0; i < bucketCount; i++){
            intermediate.add(new ArrayList<Integer>());
        }
        for(int i = 0; i < num.length; i++){
            intermediate.get((num[i] - min)/bucketSize).add(num[i]);
        }

        
        int[] res = new int[num.length];
        int j = 0;
        for(int i = 0; i < bucketCount; i++){
            int[] tmp = new int[intermediate.get(i).size()];
            int k =0;
            for (Object o : intermediate.get(i)) {
                tmp[k] = (Integer)o;
                k++;
            }
            Arrays.sort(tmp);

            for(int c = 0; c < tmp.length;c++){
                res[j] = tmp[c];
                System.out.print(res[j]+ ",");
                j++;
            }
        }

        return res;

    }


    

File: program_data/PKU_raw/22/90.c
void main()
{char a[300];
	int num[300]={0};
	int i,j,d=0,s=0,t,x,y=0;
	gets(a);
	
	for(i=0;a[i]!='\0';i++)
	{ if((a[i]>='0'&&a[i]<='9')&&(a[i+1]>='0'&&a[i+1]<='9'))
	  d=d*10+a[i]-'0';
	  else if((a[i]>='0'&&a[i]<='9')&&(a[i+1]<'0'||a[i+1]>'9'))
      {d=d*10+a[i]-'0';
	  num[s]=d;
	  s=s+1;
	  d=0;}
	}
	s=s-1;
	
	for(j=0;j<s;j++)
			for(i=0;i<s-j;i++)
			if(num[i]<num[i+1])
			{
				t=num[i];
				num[i]=num[i+1];
				num[i+1]=t;}
        for(i=0;i<=s-1;i++)
		
		for(j=1;j<=s;j++)
			if(num[j]!=num[0])
			{ x=j;  y=1;   break;}
			if(y==1)  printf("%d",num[x]);
			if(y==0)  printf("No");  

}



